### ISSUE TITLE
runner: allow execution of multiple jobs in parallel

### ISSUE THREAD

**Issue by tomaskrizek** [freeipa/freeipa-pr-ci#54]
A runner should be capable of executing multiple jobs. Initial implementation could have a predefined number of maximum concurrent jobs. In the future, each job could have the required resources associated with it. Then, the runner would be capable of fully utilizing its resources by executing as many jobs as it has resources for.

**Comment by tomaskrizek** [comment]
We've tried to implement this functionality in #114, which was based on #72. The approach was to fork the process once the task queue was generated and execute the individual tasks in their own processes. This was more difficult than expected and led to many hard-to-debug issues with communication over ssl (https://stackoverflow.com/questions/3724900/python-ssl-problem-with-multiprocessing, https://stackoverflow.com/questions/25455462/share-list-between-process-in-python-server). To avoid breaking existing functionality, we've decided to postpone this until we can implement it properly.

Ideally, generating the task queue should be completely separate from the code that is capable of executing a single task. Then, some kind of messaging should be used to communicate between the processes (e.g. [RabbitMQ](http://rabbitmq.com/)). It should be implemented in a way that allows the task queue to be generated on a different machine than the machine that will be executing the tasks. This would also enable future scaling of the system.

**Comment by tomaskrizek** [comment]
It might be possible to solve the multiprocessing issue with SSL:

```
Applications must change the PRNG state of the parent process if they use any
SSL feature with os.fork(). Any successful call of RAND_add(), RAND_bytes() 
or RAND_pseudo_bytes() is sufficient.
```

https://docs.python.org/3/library/ssl.html#multi-processing

**Comment by f-trivino** [comment]
We have several testing scenarios (jobs) where memory consumption is close to the limit. This feature will cause multiple issues when running jobs. The implementation is difficult and the added value is not significant. Also, when using different templates we will not have enough hard disk quota. We close this ticket.
